### PR TITLE
Fix webpack clean

### PR DESCRIPTION
**Pull request by JPM1118**
#127 

Moved some plugin around among the 3 Webpack configs.

Also changed the `test-windows` script.

### CODE REVIEW

File: webpack.common.js [hunk 1]
```diff
@@ -23,5 +26,13 @@ module.exports = {
   },
   plugins: [
     new CleanWebpackPlugin(),
+    new HtmlWebpackPlugin({
+      appMountId: 'app',
+      template: HtmlWebpackTemplate,
+    }),
   ],
+  output: {
+    filename: 'bundle.js',
```
Comment (JPM1118) [inline webpack.common.js]: Output needs to be here to make 'clean-webpack-plugin' have somewhere to point to for it to work in both `webpack.dev.js` and `webpack.prod.js`. At least, that's what I think is going on. It is also how it is setup in https://webpack.js.org/guides/production/ .

File: webpack.common.js [hunk 2]
```diff
@@ -1,4 +1,7 @@
+const path = require('path');
 const CleanWebpackPlugin = require('clean-webpack-plugin');
+const HtmlWebpackPlugin = require('html-webpack-plugin');
+const HtmlWebpackTemplate = require('html-webpack-template');
```
Comment (JPM1118) [inline webpack.common.js]: I added 'html-webpack-plugin' and 'html-webpack-template' here so it does not have to be added to both `webpack.dev.js` or `webpack.prod.js` . Let me know if I should not have done this.

File: server/routes/api/__test__/auth.unit.test.js [hunk 3]
```diff
@@ -48,3 +48,37 @@ describe('POST Auth/Login', () => {
...
+
+  it('will create and then save a token and expiration to user', async () => {
+    await request(app)
+      .post('/api/auth/forgot')
+      .type('form')
+      .send({ email: existingUserEmail });
+    const user = await User.findOne({ email: existingUserEmail }).exec();
+    expect(user.resetPassword.token && user.resetPassword.expiration).toBeTruthy();
+  });
+});
```
Comment (JPM1118) [inline server/routes/api/__test__/auth.unit.test.js]: Merged in from Master.

File: browser/src/react-redux/App/Nav.react.js [hunk 4]
```diff
@@ -1,24 +1,51 @@
...
+    email: PropTypes.string,
+    createdAt: PropTypes.string,
+    updatedAt: PropTypes.string,
+    __v: PropTypes.number,
+  }),
+};
+
+const mapStateToProps = state => ({ user: state.user });
+
+export default connect(mapStateToProps)(User);
```
Comment (JPM1118) [inline browser/src/react-redux/App/Nav.react.js]: Merged in from master.

File: browser/src/react-redux/App/Nav.react.js [hunk 5]
```diff
@@ -1,51 +1,24 @@
...
-    createdAt: PropTypes.string,
-    updatedAt: PropTypes.string,
-    __v: PropTypes.number,
-  }),
-};
-
-const mapStateToProps = state => ({ user: state.user });
-
-export default connect(mapStateToProps)(User);
+export default User;
```
Comment (JPM1118) [inline browser/src/react-redux/App/Nav.react.js]: Reverted master merge in.

File: server/routes/api/__test__/auth.unit.test.js [hunk 6]
```diff
@@ -48,37 +48,3 @@ describe('POST Auth/Login', () => {
     expect(res.body.email).toEqual(email); // returned email should be correct user
```
Comment (JPM1118) [inline server/routes/api/__test__/auth.unit.test.js]: reverted master merge in.

File: webpack.common.js (hunk 2 above)
Comment (benjaminsinaiko) [reply webpack.common.js]: This makes sense, `webpack.common.js` then gets merged into `dev` and `prod`.

**Pull request merged**